rules/android_binary_internal/r8.bzl

```diff
@@ -119,7 +119,10 @@ def process_r8(ctx, validation_ctx, jvm_ctx, packaged_resources_ctx, build_info_
         value = struct(
             final_classes_dex_zip = dexes_zip,
             dex_info = android_dex_info,
-            providers = [android_dex_info],
+            providers = [
+                android_dex_info,
+                AndroidPreDexJarInfo(deploy_jar),
```

Comment:
This is eventually needed here https://github.com/bazelbuild/rules_android/blob/4962fa26092de4c743457d1b566885196dc7ce99/rules/android_application/android_application_rule.bzl#L387